bokehjs/src/coffee/models/glyphs/hex_tile.ts

```diff
@@ -0,0 +1,279 @@
+import {Glyph, GlyphView, GlyphData} from "./glyph"
+
+import {PointGeometry, RectGeometry, SpanGeometry} from "core/geometry"
+import * as hittest from "core/hittest"
+import * as p from "core/properties"
+import {LineMixinVector, FillMixinVector} from "core/property_mixins"
+import {Arrayable} from "core/types"
+import {IBBox} from "core/util/bbox"
+import {Context2d} from "core/util/canvas"
+import {SpatialIndex, RBush} from "core/util/spatial"
+import {NumberSpec} from "core/vectorization"
+import {Line, Fill} from "core/visuals"
+
+import {generic_area_legend} from "./utils"
+import {Selection} from "../selections/selection"
+
+
+export interface HexTileData extends GlyphData {
+  _x: Arrayable<number>
+  _y: Arrayable<number>
+
+  sx: Arrayable<number>
+  sy: Arrayable<number>
+
+  svx: number[]
+  svy: number[]
+
+  scale: Arrayable<number>
+
+  minX: number
+  maxX: number
+  minY: number
+  maxY: number
+
+  ssize: number
+}
+
+export interface HexTileView extends HexTileData {}
+
+export class HexTileView extends GlyphView {
+  model: HexTile
+  visuals: HexTile.Visuals
+
+  scenterx(i: number): number { return this.sx[i] }
+
+  scentery(i: number): number { return this.sy[i] }
+
+  protected _set_data(): void {
+    const self = this as any
```

Comment:
Those `any`s have to be removed. Adding `_q` and `_r` (`Arrayable<number>`) to `HexTileData` should be sufficient.